pep/app/pods/read/document/route.ts

```diff
@@ -38,33 +43,57 @@ export default class ReadDocument extends PageNav(Route) {
      */
     async afterModel(model: object, transition: Transition) {
         super.afterModel(model, transition);
+        let results;
+        let resultsMeta;
+
+        // if we are transitioning from either the search results page, or another read document page
+        // attempt to pull in the existing documents models for the results list, instead of making
+        // another search query request for the same data
+        if (transition.from?.name === 'read.document' || transition.from?.name === 'search') {
+            const controller = this.controllerFor(transition.from?.name) as ReadDocumentController;
+            results = controller?.paginator?.models;
+            resultsMeta = controller?.paginator?.metadata;
+        }
 
-        const params = this.paramsFor('read.document') as ReadDocumentParams;
-        //workaround for https://github.com/emberjs/ember.js/issues/18981
-        const searchTerms = params._searchTerms ? JSON.parse(params._searchTerms) : [];
-        const facets = params._facets ? JSON.parse(params._facets) : [];
+        if (!results) {
+            const params = this.paramsFor('read.document') as ReadDocumentParams;
+            //workaround for https://github.com/emberjs/ember.js/issues/18981
+            const searchTerms = params._searchTerms ? JSON.parse(params._searchTerms) : [];
+            const facets = params._facets ? JSON.parse(params._facets) : [];
+            const cfg = this.configuration.base.search;
+            const searchParams = buildSearchQueryParams(
+                params.q,
+                searchTerms,
+                params.matchSynonyms,
+                facets,
+                params.citedCount,
+                params.viewedCount,
+                params.viewedPeriod,
+                cfg.facets.defaultFields,
+                'AND',
+                cfg.facets.valueLimit,
+                cfg.facets.valueMinCount
+            );
 
-        const searchParams = buildSearchQueryParams(
-            params.q,
-            searchTerms,
-            params.matchSynonyms,
-            facets,
-            params.citedCount,
-            params.viewedCount,
-            params.viewedPeriod
-        );
-        //if no search was submitted, don't fetch any results
-        if (hasSearchQuery(searchParams)) {
-            const controller = this.controllerFor(this.routeName);
-            const queryParams = buildQueryParams({
-                context: controller,
-                pagingRootKey: null,
-                filterRootKey: null,
-                processQueryParams: (params) => ({ ...params, ...searchParams })
-            });
+            //if no search was submitted, don't fetch any results
+            if (hasSearchQuery(searchParams)) {
+                const controller = this.controllerFor(this.routeName);
+                const queryParams = buildQueryParams({
+                    context: controller,
+                    pagingRootKey: null,
```

Comment:
While it doesn't need to be fixed for this PR, I'd like to take a look at why we need to specify `pagingRootKey` and `filterRootKey` as null almost every time we use the pagination. We should just default it so we don't have to worry about it

Reply:
`buildQueryParams` is a util function that comes from our ember-pagination addon, which has its defaults optimized for JSON-API, which have the root keys for those as `page` and `filter`, which params get nested under. I guess we could maybe create an "overridden" version of this function in PEP that proxies to the addon's function w/those set to `null`? I would still keep the addon defaults to json-api tho is thats what most of our internal-api based projects will be using moving forward.